transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebAcceptProcessor.java

```diff
@@ -179,35 +175,8 @@ protected void flushInternal(final WsebSession session) {
                     // writing data to the parent during this interim state
                     // resulting in a WriteToClosedSessionException and losing data
 
-                    // convert from session+buffer to message
-                    if (buf instanceof WsBuffer) {
-                        // reuse previously constructed message if available
-                        WsBuffer wsBuffer = (WsBuffer)buf;
-                        WsMessage wsebMessage = wsBuffer.getMessage();
-                        if (wsebMessage == null) {
-                            WsMessage newWsebMessage;
-                            if (wsBuffer.getKind() == WsBuffer.Kind.TEXT) {
-                                //if the connection is mixed transport, send textmessage
-                                newWsebMessage = new WsTextMessage(buf);
-                            }
-                            else {
-                                newWsebMessage = new WsBinaryMessage(buf);
-                            }
-
-                            if (wsBuffer.isAutoCache()) {
-                                // buffer is cached on parent, continue with derived caching
-                                newWsebMessage.initCache();
-                            }
-                            boolean wasUpdated = wsBuffer.setMessage(newWsebMessage);
-                            wsebMessage = wasUpdated ? newWsebMessage : wsBuffer.getMessage();
-                        }
-                        // flush the buffer out to the session
-                        lastWrite = flushNowInternal(writer, wsebMessage, wsBuffer, filterChain, request);
-                    }
-                    else {
-                        // flush the buffer out to the session
-                        lastWrite = flushNowInternal(writer, new WsBinaryMessage(buf), buf, filterChain, request);
-                    }
+                    // flush the message out to the session
+                    lastWrite = flushNowInternal(writer, frame, buf, filterChain, request);
```

Comment:
I think we have to do checkBuffer() for each write request in the queue, so one per execution of the loop, to make sure we issue a reconnect (downstream) as soon as we have written more that the .kb parameter